sklearn/svm/liblinear.pyx

```diff
@@ -14,7 +14,8 @@ np.import_array()
 def train_wrap(X, np.ndarray[np.float64_t,   ndim=1, mode='c'] Y,
                bint is_sparse, int solver_type, double eps, double bias,
                double C, np.ndarray[np.float64_t, ndim=1] class_weight,
-               int max_iter, unsigned random_seed, double epsilon):
+               int max_iter, unsigned random_seed, double epsilon,
+               np.ndarray[np.float64_t, ndim=1] sample_weight):
```

Comment:
adding `mode='c'` might be useful to ensure to check that the input array is contiguous and raise a `ValueError` otherwise.